### PR TITLE
Add in special way to get correct uri & label

### PR DESCRIPTION
**Pull request by lamtu1**
`FEATURE:` Add in a special way to get the URI & label correctly when rendering form on edit.

`TASK:` Task list for implementing the `label$uri` in OD2

- [x] Create `TriplePoweredService` for OD2 for functionality fetching and put as `label$uri` and create the `spec` for it
- [x] Build `Triple Powered Properties for solr doc` function in Generic Indexer
- [ ] ~(Maybe?) Need to add `FormMetadataService`~
- [x] Add the method to `generate_solr_document` to save it to Solr
- [x] Create a `label parser` class and write the `spec` for it
- [x] Add in the label functionality to `solr_document` to access the function on the variable
- [ ] (Maybe?) Add to `catalog_controller` as a `search_field`

### CODE REVIEW

**Comment by lamtu1** [comment]
fixes #2634

**Review by CGillen — CHANGES_REQUESTED** [review]
Unfortunately, this is a wider systemic issue than just `subject`, so a special case wouldn't be the best here.
This solution will also kick off a network call for each controlled term, whether to our cache or the source that's too slow for a page load. We **_have_** to do this work on index/save

You might find a better solution in placing a placeholder value in the solr index where a label couldn't be fetched on index or something else at index time. Alternatively, we could discuss a transition plan for using split label indexing like SA@OSU which might be the ultimate goal

**Comment by lamtu1** [comment]
@CGillen Right now, I can try to find a way to fix this temporary with a placeholder on index time, but from what you said, I think it is best to create a new ticket to overhaul the `uri & label` for the whole system wide like in SA@OSU which will be efficient and less broken-ness on the fetching end if they are depreciated link.

**Comment by CGillen** [comment]
I agree, lets go with a place holder `Unable to fetch label` and I'll open a ticket for transitioning to `$` split indexing

**Comment by lamtu1** [comment]
`TASK:` Task list for implementing the `label$uri` in OD2

- [ ] Create `TriplePoweredService` for OD2 for functionality fetching and put as `label$uri` and create the `spec` for it
- [ ] Build `Triple Powered Properties for solr doc` function in Generic Indexer
- [ ] (Maybe?) Need to add `FormMetadataService` if field have `Other` in `Triple Powered Properties for solr doc`
- [ ] Add the method to `generate_solr_document` to save it to Solr
- [ ] Create a `label parser` class and write the `spec` for it
- [ ] Add in the label functionality to `solr_document` to access the function on the variable
- [ ] (Maybe?) Add to `catalog_controller` as a `search_field`

File: app/services/oregon_digital/triple_powered_service.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      else
+        fetch_all_labels(uris).first
+      end
+    end
+
+    # METHOD: Functionality to fetch all labels and put it as 'label$uri'
+    def fetch_all_labels(uris)
+      labels = []
+      uris.each do |uri|
+        graph = fetch_from_store(uri)
```
Comment (CGillen) [inline app/services/oregon_digital/triple_powered_service.rb:23]: Ok, this one might be a difficult one to explain, so let me know if we need to jump on a call later.

I don't think we need the fetching part or `#predicate_labels` part of this service
Our controlled properties system is different from SA@OSU and already includes fetching. So we're able to do something like `work.subject` to give us an array of `OregonDigital::ControlledVocuabulary::Subject` objects, each with a `#fetch` method that will setup the object to have a readable label.
You should be able to get the `label$uri` setup with something like this:
```
subjects = object.subject
fetch_all_labels(subjects)

...

def fetch_all_labels(controlled_vocabs)
labels = []
controlled_vocabs.each do |cv|
  cv.fetch
  labels << "#{cv.rdf_label.first.to_s}$#{cv.rdf_uri.to_s}"
end
```

Comment (lamtu1) [reply app/services/oregon_digital/triple_powered_service.rb:23]: I will give this a try, if anything i will reach out. But thank you for the suggestion to fix this. I will get on to it.

**Comment by lamtu1** [comment]
@wickr I got a chance to talked to Corey today and I wanted to follow up with you in regards of this `label$uri` feature implementation. I remember from the last work cycle for OD2, you wanted to meet up with me and Corey to discuss in regards of this ticket. And I was wondering if you still wanted to schedule a meet up with all of us to discuss any questions or concerns you have on this ticket?

File: app/services/oregon_digital/triple_powered_service.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module OregonDigital
+  # Triple Powered Service to fetch out data from uri and grouped as 'label$uri'
+  class TriplePoweredService
```
Comment (CGillen) [inline app/services/oregon_digital/triple_powered_service.rb:5]: Maybe rename to `LabelFetchService` to better match OD2 naming scheme & `LabelParseService`
(Don't forget to rename file)

**Review by CGillen — CHANGES_REQUESTED** [review]

File: app/services/oregon_digital/triple_powered_service.rb (hunk 2 above)
Comment (lamtu1) [reply app/services/oregon_digital/triple_powered_service.rb:5]: Yes I will rename it to fit the scheme. I'm currently rebuilding the image trying to fix the `docker-compose test` but will update it once got everything to pass the test with the updated on there.

**Review by CGillen — APPROVED** [review]

**Pull request merged**